sql/parser2/eval.go

```diff
@@ -668,20 +687,41 @@ func evalFuncExpr(expr *FuncExpr, env Env) (Datum, error) {
 
 func evalCaseExpr(expr *CaseExpr, env Env) (Datum, error) {
 	if expr.Expr != nil {
-		// TODO(pmattis): These are expressions of the form `CASE <val> WHEN <val>
-		// THEN ...`.
-		return null, fmt.Errorf("TODO(pmattis): unsupported simple case expression: %T", expr)
-	}
-
-	for _, when := range expr.Whens {
-		d, err := EvalExpr(when.Cond, env)
+		// CASE <val> WHEN <expr> THEN ...
```

Comment:
is there `ELSE` as in `CASE [<val>] WHEN <[bool-]expr> THEN <expr> ELSE <expr>`?